pkg/registry/generic/registry/store.go

```diff
@@ -321,6 +369,10 @@ func (e *Store) Update(ctx api.Context, name string, objInfo rest.UpdatedObjectI
 		if err := rest.BeforeUpdate(e.UpdateStrategy, ctx, obj, existing); err != nil {
 			return nil, nil, err
 		}
+		delete := e.shouldDelete(ctx, key, obj, existing)
```

Comment:
should this be passing in the old object as well, to see if the particular update that is being made is that finalizers were non-empty, and now are empty?